Sources/Build/BuildDelegate.swift

```diff
@@ -96,12 +96,16 @@ final class TestDiscoveryCommand: CustomLLBuildCommand {
             return path.basename == "main.swift"
         }
 
+        var mainFile: AbsolutePath?
         // Write one file for each test module.
         //
         // We could write everything in one file but that can easily run into type conflicts due
         // in complex packages with large number of test targets.
         for file in outputs {
-            if isMainFile(file) { continue }
+            if isMainFile(file) {
```

Comment:
I wonder if it'll help if we check if we add a nil check here:

if mainFile == nil && isMainFile(file)

Reply:
What if there are two `main.swift` files in the test product (e.g. by mistake)? I'm afraid that `mainFile == nil && isMainFile(file)` would allow the iteration to continue for the second `main.swift` as if it was a non-main file, which could cause hard to diagnose issues.

Reply:
The main file here is the autogenerated LinuxMain.swift file. There can only ever be one of it in the output. Unfortunately, this is not represented using the type system and is an invariant.

Reply:
Ah, makes sense 👍